### PR TITLE
improvement(dev): better help text & header

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

Added some more descriptive text to the header about what can be done in the dev command.

Also sorted the command listing rendered by the `help` command by "popularity", with the primary / most commonly used commands appearing closer to the top (and putting less frequently used commands in their own section).

This should make the command listing more readable, especially for new users who aren't familiar with the available commands.

Screenshot 2023-05-17 at 22 55 51

### CODE REVIEW

**Comment by thsig** [comment]
Note: The inconsistent colours of the action durations have been fixed in https://github.com/garden-io/garden/pull/4298.

File: core/src/cli/command-line.ts [hunk 1]
```diff
@@ -486,15 +486,50 @@ ${renderDivider({ width, char, color })}
   }
 
   showHelp() {
+    // These commands are rendered first in the help text, since they're more commonly used than the others.
+    const popularCommandFullNames = new Set<string>([
```
Comment (vvagaytsev) [inline core/src/cli/command-line.ts:490]: This seems to be unused, and it duplicates the function defined in another file.

**Review by vvagaytsev — APPROVED** [review]
Left a non-blocking comment, please check it out. There is some dead code :) Otherwise looks good, thank you! 👍

**Comment by thsig** [comment]
Thanks @vvagaytsev—I think this is ready to go now.

**Review by vvagaytsev — DISMISSED** [review]
Thank you! 👍

File: core/src/commands/dev.tsx [hunk 2]
```diff
@@ -60,6 +60,9 @@ export class DevCommand extends ServeCommand<DevCommandArgs, DevCommandOpts> {
 ${renderDivider({ color: chalk.green, title: chalk.green.bold("🌳  garden dev 🌳 "), width })}
 
 ${chalk.bold(`Good ${getGreetingTime()}! Welcome to the Garden interactive development console.`)}
+
+Here, you can ${chalk.white("build")}, ${chalk.white("deploy")}, ${chalk.white("test")} and ${chalk.white("run")} anything in your project, start code syncing, stream live logs and more.
+
```
Comment (edvald) [inline core/src/commands/dev.tsx:65]: I think it'd be helpful to say a bit more, e.g. tell users about the `help` command, up/down keys to scroll through command history, tab to autocomplete etc etc.

Comment (edvald) [reply core/src/commands/dev.tsx:65]: Merging this, but imo we can improve the dev command intro message further.

**Review by edvald — APPROVED** [review]

**Review by edvald — APPROVED** [review]

**Pull request merged**